spec/release_notes_helper_spec.rb

```diff
@@ -22,6 +22,23 @@
     HEADER
   end
 
+  it 'adds a new section after any `#` comments on top' do
+    run_release_notes_test <<~HEADER
+      # This is a Markdown header
+      ## This is another kind of Markdown header
+      ### This is an H3 Markdown header
+
+    HEADER
+  end
```

Comment:
I feel like this might not be what we want, because while the `FAKE_CONTENT` from this test suite suggests that all our CHANGELOG files use `---` to underline section titles, many of our other changelogs (including [release-toolkit's own CHANGELOG](https://raw.githubusercontent.com/wordpress-mobile/release-toolkit/trunk/CHANGELOG.md)) use the `## 1.2.3` markdown syntax instead for section titles.

I'm not fully sure that our current `release_notes_helper` implementation supports using the `## x.y.z` format instead of `x.y.z\n---` to be honest, maybe it doesn't. But I could see a future where we would prefer adopting that more common `## x.y.z` markdown syntax for headers, and in such case if we allow here those to be considered comment headers, this would prevent us from allowing them to be valid `## 1.2.3` version sections in the CHANGELOG.

For that reason, I'm not sure I agree for us to consider `#`, `##` and `###` sections as being "header comments" in our CHANGELOGs.

---

An alternative that we could consider would be to use HTML comments:
 - Either to use them as header comments, i.e. you can start your `CHANGELOG.md` file with `<!--`, put some instructions, then `-->`, before starting the rest of the CHANGELOG content, and the helper would have to insert a new section for a new version only _after_ that `<!-- … -->` part
 - Or make a convention that one can use `<!-- CHANGELOG START -->` or `<!-- New Version Marker -->` or something in the CHANGELOG to mark where the helper is supposed to insert new sections in the CHANGELOG, making it even more flexible by allowing anything before that section (even `# Instructions` setions and such)

I think that solution shouldn't be too hard to implement, something like:
```ruby
line_idx = if lines.first.start_with('<!--')
             lines.find_index { |l| l.end_with('-->') } &+ 1
           else
             lines.find_index { |l| l.chomp == '<!-- INSTRUCTION END MARKER -->' } &+ 1
           end
line_idx ||= lines.finx_index { |l| !l.start_with('***') && !l.start_with('//') && !l.chomp.empty? }
```

That way, if the file starts with `<!--` then we will start inserting on the line after the closing `-->`, otherwise we'll try to find the special line with content `<!-- INSTRUCTION END MARKER -->` if it exist (and if so, start inserting after it), and finally if none of those found anything, we fall back to the current behavior of inserting after any first lines starting with `//` or `***`.

Reply:
> An alternative that we could consider would be to use HTML comments:

I really like this approach! Seems much simpler and easy to just have anything above a `<!-- CHANGELOG START -->` be considered the file header and to insert the new section after that.